proposals/0010-resource-element-type-validation.md

```diff
@@ -184,7 +184,7 @@ struct vector_type_info<T __attribute__((ext_vector_type(N)))> {
   constexpr static unsigned Size = N;
 };
 
-bool is_spirv_target = ...;
+const bool is_spirv_target = getASTContext().getTargetInfo().getTriple().isSPIRV();

```

Comment:
I'm confused - is this code sample HLSL code, or part of the Clang implementation?

Reply:
It would be part of the clang implementation, C++ code within HLSLExternalSemaSource.cpp. I'll make that clearer.

Reply:
Apart from line 185, this looks like what we'd write in hlsl.h, but because it isn't all currently spellable we'd have to implement it in HLSLExternalSemaSource.cpp by constructing AST nodes.

Reply:
We could define such a thing in the HLSL headers by doing something like:

```c
#if __is_target_arch(spirv)
constexpr bool is_spirv_target = true;
#else
constexpr bool is_spirv_target = false;
#endif
```

In the External Sema Source we can just generate the constraints differently based on the target.